python/foglamp/tasks/north/sending_process.py

```diff
@@ -683,22 +704,32 @@ def send_data(self, stream_id):
         Args:
         Returns:
         Raises:
-        Todo:
         """
         SendingProcess._logger.debug("{0} - ".format("send_data"))
         try:
             start_time = time.time()
             elapsed_seconds = 0
+
             while elapsed_seconds < self._config['duration']:
+
                 try:
                     data_sent = self._send_data_block(stream_id)
                 except Exception as e:
                     data_sent = False
                     _message = _MESSAGES_LIST["e000021"].format(e)
                     SendingProcess._logger.error(_message)
+
+                # Terminates the execution in case a signal has been received
+                if SendingProcess._stop_execution:
```

Comment:
I believe this should move to top in `while` before `data_sent = self._send_data_block(stream_id)`

Reply:
done